src/test/java/org/opensearch/knn/integ/BinaryIndexIT.java

```diff
@@ -0,0 +1,242 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.knn.common.Constants.FIELD_FILTER;
+import static org.opensearch.knn.common.Constants.FIELD_TERM;
+import static org.opensearch.knn.common.KNNConstants.ENCODER_SQ;
+import static org.opensearch.knn.common.KNNConstants.K;
+import static org.opensearch.knn.common.KNNConstants.KNN;
+import static org.opensearch.knn.common.KNNConstants.MAX_DISTANCE;
+import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
+import static org.opensearch.knn.common.KNNConstants.MIN_SCORE;
+import static org.opensearch.knn.common.KNNConstants.PATH;
+import static org.opensearch.knn.common.KNNConstants.QUERY;
+import static org.opensearch.knn.common.KNNConstants.TYPE_NESTED;
+import static org.opensearch.knn.common.KNNConstants.VECTOR;
+
+/**
+ * This class contains integration tests for binary index with HNSW in Faiss
+ */
+@Log4j2
+public class BinaryIndexIT extends AutoIndexCleanUpTestCase {
+    private static TestUtils.TestData testData;
+
+    @BeforeClass
+    public static void setUpClass() throws IOException {
+        if (BinaryIndexIT.class.getClassLoader() == null) {
+            throw new IllegalStateException("ClassLoader of BinaryIndexIT Class is null");
+        }
+        URL testIndexVectors = BinaryIndexIT.class.getClassLoader().getResource("data/test_vectors_binary_1000x128.json");
+        URL testQueries = BinaryIndexIT.class.getClassLoader().getResource("data/test_queries_binary_100x128.csv");
+        URL groundTruthValues = BinaryIndexIT.class.getClassLoader().getResource("data/test_ground_truth_binary_100.csv");
+        assert testIndexVectors != null;
+        assert testQueries != null;
+        assert groundTruthValues != null;
+        testData = new TestUtils.TestData(testIndexVectors.getPath(), testQueries.getPath(), groundTruthValues.getPath());
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_whenKnnIndexIsTrue_thenCreateIngestQueryWorks() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        createKnnHnswBinaryIndex(true, KNNEngine.FAISS, indexName, fieldName, 16, null);
+
+        // Ingest
+        Byte[] vector1 = { 0b00000001, 0b00000001 };
+        Byte[] vector2 = { 0b00000011, 0b00000001 };
+        Byte[] vector3 = { 0b00000111, 0b00000001 };
+        Byte[] vector4 = { 0b00001111, 0b00000001 };
+        addKnnDoc(indexName, "1", fieldName, vector1);
+        addKnnDoc(indexName, "2", fieldName, vector2);
+        addKnnDoc(indexName, "3", fieldName, vector3);
+        addKnnDoc(indexName, "4", fieldName, vector4);
+
+        // Query
+        float[] queryVector = { (byte) 0b10001111, (byte) 0b10000000 };
+        int k = 4;
+        Response searchResponse = searchKNNIndex(indexName, new KNNQueryBuilder(fieldName, queryVector, k), k);
```

Comment:
Use XcontentBuilder here as well?

Reply:
Why not KNNQueryBuilder?

Reply:
KNNQueryBuilder is an internal implementation, is that to right to use in integ test? we are creating interdependencies with actual code. 

Ideally we should just be passing a json or use a java client to do so.

Reply:
That make sense.